src/backend/gpopt/translate/CTranslatorRelcacheToDXL.cpp

```diff
@@ -652,6 +653,10 @@ CTranslatorRelcacheToDXL::Pmdrel
 				erelstorage = IMDRelation::ErelstorageAppendOnlyParquet;
 			}
 		}
+
+		// get number of leaf partitions
+		ulLeafPartitions = gpdb::INumberPartitions(oid);
+
```

Comment:
```cpp
ulLeafPartitions = 0;
if (gpdb::FRelPartIsRoot(oid))
{
   ulLeafPartitions = gpdb::UlLeafPartitions(oid);
}
```

I recommend we change to this code snippet, thus we can get rid of INumberPartitions.